### ISSUE TITLE
SPIKE: Determine VBA Regional Office IA implementation

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#15752]
## Problem Statement

The official names of VBA facilities can be very lengthy. eg:

- Honolulu VA Regional Benefit Office at Spark M. Matsunaga Department of Veterans Affairs Medical Center
- VA Regional Benefit Satellite Office at Samuel S. Stratton Department of Veterans Affairs Medical Center

[Previous conversation with CAIA](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15490#issuecomment-1758616883) led us to look at [ways to shorten VBA facility names](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/facilities/naming-schema#vba-benefits-facilities-naming-schema) so we don't end up with incredibly long URLS / H1s / breadcrumbs for users.

We developed [designs](https://www.sketch.com/s/891d33ae-152d-471c-ab5d-9aedf89cf6ff) for splitting off Shared Location into text associated with the H1 so it wouldn't clog up the H1/URL/Breadcrumb. 

## VA Regional Benefit Offices
This approach will work very well for VA Regional Benefit Offices which almost always have a unique region as part of their official names.

- **Official name:** Honolulu VA Regional Benefit Office at Spark M. Matsunaga Department of Veterans Affairs Medical Center
- **H1/breadcrumb:** Honolulu VA Regional Benefit Office
- **URL:** va.gov/honolulu-va-regional-benefit-office
- **Text associated with H1:** Located at Spark M. Matsunaga Department of Veterans Affairs Medical Center

## VA Regional Benefit Satellite Offices
However, it won't work as well for VA Regional Benefit Satellite Offices, since they don't currently have any unique identifiers aside from their shared location:

- **Official name:** VA Regional Benefit Satellite Office at Samuel S. Stratton Department of Veterans Affairs Medical Center
- **H1/breadcrumb:** VA Regional Benefit Satellite Office 
- **URL:** va.gov/va-regional-benefit-satellite-office ❌
-  **URL:** va.gov/va-regional-benefit-satellite-office-at-samuel-s-stratton-department-of-veterans-affairs-medical-center 
- **Text associated with H1:** Located at Samuel S. Stratton Department of Veterans Affairs Medical Center

We may need to specify a REGION prefix for Satellite Offices. The [VBA naming proposal](https://dvagov.sharepoint.com/:x:/s/SitewideFacilitiesTeam/ESGDTL0pkqdDiKT3oqcFoZEBuEtSx0NdGZIbvUwY7TwQ9w?e=9amPaQ) contains extrapolated Regions for all Satellite Offices

- **Official name:** VA Regional Benefit Satellite Office at Samuel S. Stratton Department of Veterans Affairs Medical Center
- **Region:** Albany
- **H1/breadcrumb:** Albany VA Regional Benefit Satellite Office 
- **URL:** va.gov/albany-va-regional-benefit-satellite-office
- **Text associated with H1:** Located at Samuel S. Stratton Department of Veterans Affairs Medical Center

### Only distinguished by Shared Location

Even with the addition of a region prefix, there are some instances where that might not be enough to differentiate the urls for satellite offices

- **Official name:** VA Regional Benefit Satellite Office at Maui VA Clinic
- **Official name:** VA Regional Benefit Satellite Office at Maui Vet Center
- **Region:** Maui
- **URL:** va.gov/maui-va-regional-benefit-satellite-office ❌
- **URL:** va.gov/maui-va-regional-benefit-satellite-office ❌

Do we keep these instances at the long URL of the current official name?
- **URL:** va.gov/va-regional-benefit-satellite-office-at-maui-va-clinic
- **URL:** va.gov/va-regional-benefit-satellite-office-at-maui-vet-center

Do we put the shared location up first in the URL?
- **URL:** va.gov/maui-va-clinic-va-regional-benefit-satellite-office
- **URL:** va.gov/maui-vet-center-va-regional-benefit-satellite-office

Though that then just begs the question why these aren't structured as sub facilities:
- **URL:** va.gov/maui-vet-center/va-regional-benefit-satellite-office

### Shared Location is actually a Region

- **Official name:** VA Regional Benefit Satellite Office at Fresno
- **H1/breadcrumb:** VA Regional Benefit Satellite Office
- **Text associated with H1:** Located at Fresno ❌

### Region may need to include city and state/territory

- VA Regional Benefit Satellite Office at Arlington, TX
- VA Regional Benefit Satellite Office at Fort Buchanan Puerto Rico

### Do we need style guidance for military bases?

- VA Regional Benefit Satellite Office at Marine Corps Base Camp Pendleton
- VA Regional Benefit Satellite Office at Naval Submarine Base New London

## Next steps
We need to make a decision on the exact way VBA Regional Offices and Satellite office will be named on the website & the technical implementations across all places the data is stored (Sandy's DB, Drupal, Lighthouse)
- Are we keeping the word regional in the name of all satellite offices? Even though it it dilutes the distinction between regional and satellite offices?

## Documents
- [IA spec from CAIA](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/information-architecture/ia-design-docs/vba-benefit-offices-migration.md)
- [VBA naming schema](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/facilities/naming-schema#vba-benefits-facilities-naming-schema)
- [Sharepoint VBA naming proposal](https://dvagov.sharepoint.com/:x:/s/SitewideFacilitiesTeam/ESGDTL0pkqdDiKT3oqcFoZEBuEtSx0NdGZIbvUwY7TwQ9w?e=9amPaQ)
- [Working mural](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?sender=u9c899abc4fe36c9a698b1647)
- [Sketch file](https://www.sketch.com/s/891d33ae-152d-471c-ab5d-9aedf89cf6ff)
- [CAIA ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/64934)
- [Previous ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15490)

## Acceptance Criteria - Sprint 103 (5 points)
- [x] Document proposal for splitting current official name field into parts
- [x] Proposal should cover H1s, Breadcrumbs, URLs, Drupal Sections, etc

## Acceptance Criteria - Sprint 104 (8 points)
- [x] Proposal should cover technical implementation in Drupal, Sandy's DB, and Facilities API
- [x] Review proposal with Engineering, Product, and CAIA (if needed)
- [x] Once proposal is approved, update [Naming Schema](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/facilities/naming-schema#vba-benefits-facilities-naming-schema) and update any follow up tickets
- [ ] ~~Submit Collab Cycle Analytics request per https://github.com/department-of-veterans-affairs/va.gov-team/issues/46029~~ Max will finish VBA analytics ticket, add any comments to the collab cycle analytics request, and then tag Jane to submit the request.

**Comment by davidmpickett** [comment]
Notes from quick convo @mmiddaugh and I had this week:
- Short names proposal might not be best thing to broach with VBA national stakeholders immediately while reengaging them in project
- Need to find a time to talk about technical options with engineering. Can we use something for URL and Breadcrumbs while having something different for H1? What concerns does that raise?

**Comment by davidmpickett** [comment]
Updates from 16th minute discussion in scrum today:

> - Short names proposal might not be best thing to broach with VBA national stakeholders immediately while reengaging them in project

@mmiddaugh reported that the first reengagement meeting with VBA stakeholders went really well, so maybe there's still a chance that we would have trust and buy in from them to work on this

- Need to find a time to talk about technical options with engineering. Can we use something for URL and Breadcrumbs while having something different for H1? What concerns does that raise?

From Steve - It is possible to have 2 fields on a node, and have those fields power different aspects of the iA (breadcrumb/URL) and another field that is the H1 (node title). If we are going to have 2 different strings, they should be 2 different fields, vs. trying to programmatically process a single field to get 2 different strings.

cc @jilladams for visibility

**Comment by davidmpickett** [comment]
Noting that @thejordanwood created designs to account for complex VBA names: https://www.sketch.com/s/891d33ae-152d-471c-ab5d-9aedf89cf6ff

**Comment by davidmpickett** [comment]
[VBA naming proposal jan 17.xlsx](https://github.com/department-of-veterans-affairs/va.gov-cms/files/13970199/VBA.naming.proposal.jan.17.xlsx)
Static copy of Sharepoint doc for ease of access

**Comment by xiongjaneg** [comment]
Other notes from [Oct. 31 16th minute](https://trello.com/c/muABnFi3/245-urls-breadcrumbs-for-vba-based-on-naming):
Another aspect: distinction between H1 / breadcrumbs.

Can a short slug for URL / longer full title H1 be a thing?

IA governance would need to sign off BUT: a name is a name, so they can’t really speak to that. IA has supported shorter URLs if possible.

Technical problem re: whether / how we can pull off:

Full H1
Shorter URL
Specified breadcrumb
All breadcrumbs are in Drupal, so this is in Drupal camp to resolve.

**Comment by davidmpickett** [comment]
Updated static copy of spreadsheet. Now with better colors and sorting! [VBA naming proposal jan 18.xlsx](https://github.com/department-of-veterans-affairs/va.gov-cms/files/13980535/VBA.naming.proposal.jan.18.xlsx)

**Comment by mmiddaugh** [comment]
Imagine a future state in which a benefit location co-located within a medical facility or Vet Center could be listed as a service location within the service accordion on the VAMC or Vet Center page. 
Example: VA Regional Benefit Satellite Office (at Des Moines VA Medical Center) is listed as a service location for Returning service member care on Des Moines VA Medical Center.
Is there anything we could/should do now from a content model/data structure perspective to set us up better for that future?

**Comment by davidmpickett** [comment]
> Imagine a future state in which a benefit location co-located within a medical facility or Vet Center could be listed as a service location within the service accordion on the VAMC or Vet Center page. Example: VA Regional Benefit Satellite Office (at Des Moines VA Medical Center) is listed as a service location for Returning service member care on Des Moines VA Medical Center. Is there anything we could/should do now from a content model/data structure perspective to set us up better for that future?

This is pretty much the opposite of how the content model is currently structured and there may be a point in the future where we basically need restructure everything to make Service Locations the focal point rather than Facilities. (I like to think about it like when you realize a sock is inside out and you need to pull it through itself to hide the seams. It still works as a sock either way, but one way is slightly more comfortable and stylish.)

**Current**
- Facility A
  - Facility Service 1
    - Service Location a
  - Facility Service 2
    - Service Location b
- Facility B
  - Facility Service 3
     - Service location c

Right now Service Locations are Paragraph Types rather than Content Types. This means that a Service Location doesn't really exist on it's own independent of the Facility Service it is attached to. It makes is very hard for a Facility Service 
(Returning service member care at Des Moines VA Medical Center) to reference a Service Location attached to a different Facility Service at a different facility (Returning service member care at VA Regional Benefit Satellite Office at Des Moines VA Medical Center).

We do have a field on VBA Facilities for Shared VHA Location that could theoretically allow queries to go up, over, and down to pull this information from a Service Location. This is definitely something we want to preserve, but we aren't currently using it to power anything.

![Screenshot 2024-01-18 161747](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/dab66dd1-6467-4da2-b4f2-c39a9a001597)

I do think that there is a future where we create a "Reusable Service Location" Content Type similar to how Public Websites has a distinction between Reusable Q&As (Content Type) and Page-specific Q&As (Paragraph Type). 

- Facility A
  - Facility Service 1
    - Facility-specific Service Location a
    - Reference to Reusable Service Location C
- Facility B
  - Facility Service 2
    - Reference to Reusable Service Location C
- Reusable Service Location C

Another possibility is that we update Service Location fields like hours and address to have "Choose a different facility to reference" instead of only referencing the facility that they are nested under.

**Comment by davidmpickett** [comment]
> Is there anything we could/should do now from a content model/data structure perspective to set us up better for that future?

@mmiddaugh I realize I just said a lot of words without actually answering your question😅 Here's a succinct actual response:

This future is exactly one of the reasons we are putting so much time and energy into having VAMCs and VBA both Service Location. By storing the most vital and reusable elements in the exact same format it will make cross-pollination possible. By contrast, Vet Centers don't currently use Service Locations on their Facility Services, so a future where a VBA Service Location is shown on a Vet Center Service is further off.

**Comment by xiongjaneg** [comment]
Dave has another ticket address the long names with "and" vs. "at."

**Comment by davidmpickett** [comment]
> We need to make a decision on the exact way VBA Regional Offices and Satellite office will be named on the website & the technical implementations across all places the data is stored (Sandy's DB, Drupal, Lighthouse)
> 
> Are we keeping the word regional in the name of all satellite offices? Even though it it dilutes the distinction between regional and satellite offices?

Chatted with @mmiddaugh about proposing the official label of satellite offices
- **From:** VA Regional Benefit Satellite Office
- **To:** VA Satellite Benefit Office

She will discuss with VBA stakeholders

**Comment by davidmpickett** [comment]
[VBA naming proposal](https://dvagov.sharepoint.com/:x:/s/SitewideFacilitiesTeam/ESGDTL0pkqdDiKT3oqcFoZEBuEtSx0NdGZIbvUwY7TwQ9w?e=9amPaQ) has been updated with columns for URLs

![Screenshot 2024-02-01 at 11 44 53 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/b7dbf96e-67dd-4ba7-8082-207c516e5ec0)

**Comment by davidmpickett** [comment]
Mural: https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?sender=u9c899abc4fe36c9a698b1647

**Comment by davidmpickett** [comment]
End of sprint update - 5 points completed. 8 points remaining for next sprint. I will also need some of that capacity to come from Drupal & Front End. @xiongjaneg we can chat more in our sync tomorrow

**Comment by davidmpickett** [comment]
Meeting to discuss initial draft of plan with engineers scheduled for 2/14

**Comment by davidmpickett** [comment]
[VBA naming proposal - feb 14 2024.xlsx](https://github.com/department-of-veterans-affairs/va.gov-cms/files/14285022/VBA.naming.proposal.-.feb.14.2024.xlsx)
Static copy of the sharepoint spreadsheet for convenience

**Comment by davidmpickett** [comment]
## Agenda for 2/14
- Introduction to problem space and important documents - (5 min)
  - [This github ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15752)
  - [Working mural](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?sender=u9c899abc4fe36c9a698b1647)
  - Naming proposal spreadsheet ([sharepoint](https://dvagov.sharepoint.com/:x:/s/SitewideFacilitiesTeam/ESGDTL0pkqdDiKT3oqcFoZEBuEtSx0NdGZIbvUwY7TwQ9w?e=9amPaQ) = source of truth) ([static copy](https://github.com/department-of-veterans-affairs/va.gov-cms/files/14285022/VBA.naming.proposal.-.feb.14.2024.xlsx) for convenience)
  - [Facilities naming schema](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/facilities/naming-schema#vba-benefits-facilities-naming-schema)
- [Overview of the problem](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707769431021) - (10 min)
- [Shared location details and migration proposal](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707789255822) - (10 min)
- [Office type details](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707791846562) - (10 min)
- [Geographic region](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707792029497) - (10 min)
- [Condensed name - migration options](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707789838301) - (10 min)

**Comment by jilladams** [comment]
Updated ACs in this ticket based on Dave's end of sprint 103 update. @davidmpickett I'd like to try to follow the UX paradigm we've used on other UX tickets, to spell out what we can commit to per sprint in terms of ACs / points. I think that'll help us get a grip on whether we're overplanning for you.

**Comment by davidmpickett** [comment]
WORK IN PROGRESS

## Phase 1 - Implement "shared location"

Based on conversations on 2/14 and subsequent discovery, I am proposing [implementing Shared Location](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707787948375) as the first phase of the renaming effort.

![VBA Naming Logic_2024-02-20_21-07-00](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/58d69d08-21dc-4ffb-bc26-287a126adf79)

### Ticket 1 - Sandy's DB clean up
- Audit the fields `Non-VA location official name`, `Shared VHA location`, and `Official name`
- Update the fields `Non-VA location official name` and `Shared VHA location` to match reality
- Do not update `official name` unless it contains ontologically inaccurate information. This is primarily an additive audit,

**Comment by davidmpickett** [comment]
## Thread 2 - Adding "Geographic Regions" to Satellite Offices

We need to specify a REGION prefix for Satellite Offices. The [VBA naming proposal](https://dvagov.sharepoint.com/:x:/s/SitewideFacilitiesTeam/ESGDTL0pkqdDiKT3oqcFoZEBuEtSx0NdGZIbvUwY7TwQ9w?e=9amPaQ) contains extrapolated Regions for all Satellite Offices.

![VBA Naming Logic_2024-02-22_03-12-09](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/ed1261af-6acf-4e21-b7ba-91d09ac9feb6)

### Ticket 1 - Confirm "geographic region" approach with VBA
- Product
- Michelle to chat with stakeholders and confirm the overall approach 
- Share proposed geographic regions for confirmation
- For the 10 locations that are only distinguished by shared location 1) confirm they are all active locations 2) decide on how to differentiate any remaining conflicts
- Could be pared down from the full list of ALL regional offices and satellite offices to just the ones for MVP if we need to narrow focus
- Blocks ticket 3

### Ticket 2 - Reuse name field or create new field
- Product / Drupal Engineering / Lighthouse?
- Determine the [best approach for changing names](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1707266192237/97f7666116d3fa72ccfe4b136a23a7f1e948432b?wid=0-1707789838301)
- Option 1 - Create "Condensed name" as new field in Sandy's DB
- Option 2 - Overwrite official name in Sandy's DB to condensed name
- Option 3 - Create condensed name as new field in Drupal 
- If we go with option 1 or 3, are we inserting geographic regions into condensed name or official name?
- We briefly chatted about these paths on 2/14, but it was at the tail end of a long meeting.
- Might block ticket 3
- Might make ticket 7 in thread 1 unnecessary

### Ticket 3 - Update names in Sandy's DB (or Drupal?)
- Blocked by ticket 1 & maybe 2
- Product / Drupal Engineering
- Once geographic regions are approved, add them into the names that need it (all satellite offices + Manilla)
- This ticket may or may not happen at the same time at Ticket 7 in Thread 1.

**Comment by davidmpickett** [comment]
@mmiddaugh based on our conversation today, I started a draft of an Analytics Request with URLs for the Regional Offices that are part of the MVP and have a name that won't change (all of them except Honolulu and Anchorage). That seems like a good batch to focus on for staging review and more analytic request can come later in the process.

@xiongjaneg or @jilladams, I could use some help with filling out the other aspects of that Analytics Request.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/76780

**Comment by davidmpickett** [comment]
I have a pull request for updating the naming schema documentation. 
https://github.com/department-of-veterans-affairs/va.gov-team/pull/76781

**Comment by mmiddaugh** [comment]
For the engineering conversation, there are four locations which have shared vha locations not yet reflected in the name

Current name | vha shared location id | vha shared location name 
---| --- | --- | 
| White River Junction VA Regional Benefit Office | vha_405 | at White River Junction VA Medical Center |
| Cheyenne VA Regional Benefit Office | vha_442 | at Cheyenne VA Medical Center
| Muskogee VA Regional Benefit Office | vha_623 | at Jack C. Montgomery Department of Veterans Affairs Medical Center
| San Diego VA Regional Benefit Office | vha_664BY | at Kearny Mesa VA Clinic

**Comment by davidmpickett** [comment]
@jilladams @xiongjaneg Re the last AC

> Submit Collab Cycle Analytics request per Collaboration Cycle for Sitewide Facilities VBA Regional Offices  va.gov-team#46029

- I created a draft: https://github.com/department-of-veterans-affairs/va.gov-team/issues/76780
- Might make sense for @maxx1128 to finish and submit as part of #16311

**Comment by jilladams** [comment]
Closing based on notes in comments and the follow tickets that are created / queued. If anyone has concerns please feel free to flag.
